src/realm/binary_data.hpp

```diff
@@ -77,6 +77,12 @@ class BinaryData {
     {
         return m_size;
     }
+    void remove_zero_term() noexcept
+    {
```

Comment:
I'd feel more comfortable if either the following asserts were REALM_ASSERT_RELEASE, or this method handled null and did a no-op for that case

Reply:
Fixed